http-builder-ng-core/src/main/java/groovyx/net/http/JavaHttpBuilder.java

```diff
@@ -92,58 +109,84 @@ private PasswordAuthentication getAuthInfo() {
+                    https.setSSLSocketFactory(sslContext.getSocketFactory());
+                }
 
-                    JavaToServer j2s = null;
-                    if (cr.actualBody() != null) {
-                        j2s = new JavaToServer();
-                        requestConfig.findEncoder().accept(requestConfig, j2s);
-                    }
+                final ChainedHttpConfig.ChainedRequest cr = requestConfig.getChainedRequest();
 
-                    addHeaders();
+                JavaToServer j2s = null;
+                if (cr.actualBody() != null) {
+                    j2s = new JavaToServer();
+                    requestConfig.findEncoder().accept(requestConfig, j2s);
+                }
 
-                    connection.connect();
+                if(log.isDebugEnabled()){
+                    log.debug("Request-URI: {}", theUri);
+                }
+
+                addHeaders();
 
-                    if (j2s != null) {
-                        j2s.transfer();
+                connection.connect();
+
+                if (j2s != null) {
+                    if (contentLog.isDebugEnabled()) {
+                        contentLog.debug("Request-Body: {}", j2s.content());
                     }
 
-                    return handleFromServer();
-                });
+                    j2s.transfer();
+
+                }
+
+                final JavaFromServer fromServer = new JavaFromServer(theUri);
+
+                if (contentLog.isDebugEnabled()) {
+                    contentLog.debug("Response-Body: {}", fromServer.content());
+                }
+
+                if (headerLog.isDebugEnabled()) {
+                    fromServer.getHeaders().forEach(header -> headerLog.debug("Response-Header: {} -> {}", header.getKey(), header.getValue()));
+                }
+
+                return HANDLER_FUNCTION.apply(requestConfig, fromServer);
+            });
         }
 
         protected class JavaToServer implements ToServer {
 
-            private InputStream inputStream;
+            private BufferedInputStream inputStream;
 
             public void toServer(final InputStream inputStream) {
-                this.inputStream = inputStream;
+                this.inputStream = inputStream instanceof BufferedInputStream ? (BufferedInputStream) inputStream : new BufferedInputStream(inputStream);
             }
 
             void transfer() throws IOException {
-                NativeHandlers.Parsers.transfer(inputStream, connection.getOutputStream(), true);
+                IoUtils.transfer(inputStream, connection.getOutputStream(), true);
+            }
+
+            public String content() {
+                try {
+                    return IoUtils.copyAsString(inputStream);
+                } catch (IOException ioe) {
+                    log.warn("Unable to render request stream due to error (may not affect actual content)", ioe);
```

Comment:
Nitpick, but let's guard all logging statements for consistency. I know I will add some string interpolation crap here and then forget to add the guards.